node/repo/memrepo.go

```diff
@@ -106,7 +106,7 @@ func (lmem *lockedMemRepo) Path() string {
 		panic(err) // only used in tests, probably fine
 	}
 
-	if lmem.t == StorageMiner {
+	if lmem.t.Type() == "StorageMiner" || lmem.t.Type() == "Boost" {
```

Comment:
I'd use trait-interfaces instead of hard-coding (so this would be something like `if lmem.t.(interface{HasStorage()})`, and maybe a separate trait for `deal-staging`)

Reply:
Check last commit, and let me know if this is what you had in mind: https://github.com/filecoin-project/lotus/pull/8086/commits/9fd7e71a07c0772d1a196c4a53fa72140dacf189